plotly/tools.py

```diff
@@ -191,30 +198,46 @@ def set_config_file(plotly_domain=None,
     settings = get_config_file()
     if isinstance(plotly_domain, six.string_types):
         settings['plotly_domain'] = plotly_domain
+    elif plotly_domain is not None:
+        raise TypeError('Input should be a string')
     if isinstance(plotly_streaming_domain, six.string_types):
         settings['plotly_streaming_domain'] = plotly_streaming_domain
+    elif plotly_streaming_domain is not None:
+        raise TypeError('Input should be a string')
     if isinstance(plotly_api_domain, six.string_types):
         settings['plotly_api_domain'] = plotly_api_domain
+    elif plotly_api_domain is not None:
+        raise TypeError('Input should be a string')
     if isinstance(plotly_ssl_verification, (six.string_types, bool)):
         settings['plotly_ssl_verification'] = plotly_ssl_verification
+    elif plotly_ssl_verification is not None:
+        raise TypeError('Input should be a boolean')
     if isinstance(plotly_proxy_authorization, (six.string_types, bool)):
         settings['plotly_proxy_authorization'] = plotly_proxy_authorization
+    elif plotly_proxy_authorization is not None:
+        raise TypeError('Input should be a boolean')
+    if isinstance(world_readable, bool):
+        settings['world_readable'] = world_readable
+        kwargs = {'world_readable': world_readable}
+        session.update_session_plot_options(**kwargs)
+    elif world_readable is not None:
+        raise TypeError('Input should be a boolean')
     utils.save_json_dict(CONFIG_FILE, settings)
     ensure_local_plotly_files()  # make sure what we just put there is OK
 
 
-def get_config_file(*args):
-    """Return specified args from `~/.plotly_credentials`. as dict.
+def get_config_file(**kwargs):
```

Comment:
why should this be `kwargs` and not `args`? i'd expect to call this like:

`get_config_file(['plotly_domain', 'plotly_api_domain'])`

Or

`get_config_file()['plotly_domain']`